packages/aws-cdk-lib/aws-ecs/lib/cluster.ts

```diff
@@ -474,6 +475,10 @@ export class Cluster extends Resource implements ICluster {
   }
 
   private configureAutoScalingGroup(autoScalingGroup: autoscaling.AutoScalingGroup, options: AddAutoScalingGroupCapacityOptions = {}) {
+    const enableEcsAddSecurityGroup = FeatureFlags.of(this).isEnabled(cxapi.ECS_ADD_SECURITY_GROUP_TO_ASG_CAPACITY_PROVIDERS);
```

Comment:
I'm not sure this needs a feature flag. Can you explain what is the breaking change here?

Reply:
https://github.com/aws/aws-cdk/pull/24316
This is the actually ticket which contain all the conversation which is abandoned due to no review or no updates received for this for past 4 weeks.

Without providing feature flag the integ tests present in ecs-patterns are breaking. As this is a p1 fix but it may be breaking for other customers-- we asked @TheRealAmazonKendra is there any guidance on whether we need to add a feature flag here.

She said this need feature flag. After I created a feature flag the integ test are not breaking in other modules

Reply:
I see in that PR that breaking changes were mentioned, but I don't see any discussion of what those breaking changes were. Can you list them here?

Reply:
This are breaking in aws-ecs-patterns at that time 
<img width="1536" alt="image" src="https://user-images.githubusercontent.com/115483524/234064460-d94d6355-0136-43c4-94da-9cc88b3715a9.png">

Reply:
But is that actually a breaking change? Just because the template changed doesn't mean it would break anyone.

Reply:
When we create a new feature flag this will affect the behavior of the tests in modules which depend on aws-ecs. So the cached snapshot will not have the feature flag enabled, but the new snapshot will.
The alb-ecs-service-command-entry-point test which is failing in aws-ecs-patterns  due to a weird regression. the cached snapshot doesn’t deploy, possibly due to a mutation of the image or a change in ecs behavior. it was odd; neither Austin nor Yurui could really figure out what happened it seems. but finally Yurui supplied Austin with the new command/entry-point to use where this was a particularly tricky one it seems. and fix it by running yarn integ --update-on-failed --disable-update-workflowto make sure that the new snapshot deployed.


But without feature flag all the integ test in aws-ecs-patterns which contain autoscaling got break at that time. I dont have the exact screenshot related that as per Kendra Neil guidence I created a feature flag to solve this one